core/LSU.cpp

```diff
@@ -643,43 +685,28 @@ namespace olympia
         return false;
     }
 
-
-    // Access MMU/TLB
-    bool LSU::MMULookup_(const MemoryAccessInfoPtr & mem_access_info_ptr)
+    // Append new load instruction into issue queue
+    void LSU::appendLoadQueue_(const LoadStoreInstInfoPtr & inst_info_ptr)
     {
-        const InstPtr & inst_ptr = mem_access_info_ptr->getInstPtr();
-        uint64_t vaddr = inst_ptr->getTargetVAddr();
+        sparta_assert(load_queue_.size() <= ld_queue_size_,
+                      "Appending load queue causes overflows!");
 
-        bool tlb_hit = false;
+        // Always append newly dispatched instructions to the back of load queue
+        load_queue_.push_back(inst_info_ptr);
```

Comment:
Yeah, you can do this:
```suggestion
        inst_info_ptr->storeLQPosition(load_queue_.push_back(inst_info_ptr));
```

Reply:
I will remove the code from this PR to keep the changes related to the separation of the cache and mmu